ui/src/main/webapp/wizards/sources/index.js

```diff
@@ -23,15 +23,16 @@ const WizardView = ({ id, children, clearWizard }) => (
 
 const Wizard = connect(null, { clearWizard })(WizardView)
 
+const stageMapping = {
+  welcomeStage: WelcomeStage,
+  discoveryStage: DiscoveryStage,
+  sourceSelectionStage: SourceSelectionStage,
+  confirmationStage: ConfirmationStage,
+  completedStage: CompletedStage
+}
+
 let StageRouter = ({ stage, messages }) => {
-  const stageMapping = {
-    welcomeStage: <WelcomeStage />,
-    discoveryStage: <DiscoveryStage />,
-    sourceSelectionStage: <SourceSelectionStage />,
-    confirmationStage: <ConfirmationStage />,
-    completedStage: <CompletedStage />
-  }
-  return (stageMapping[stage])
+  return React.createElement(stageMapping[stage])
```

Comment:
Every time `StageRouter` was called, it would recreate the entirety of `const stageMapping` with all of it's stages. This way it only creates the react component it needs one time.